### PR TITLE
[bphh-1682] hotfix to regenerate missing pdfs on user view and push websocket update

### PR DESCRIPTION
**Pull request by taw27**
## Description
Right now there is no error handling to retry failed generation of pdf files.  This is due a limitation of the architecture as we are using a node process to generate the files

To mitigate the problem, this PR implements a hotfix to regenerate any missing pdfs, when a user opens the submiteted permit application

## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [x ] 🔥 Hotfix
- [ ] 🐛 Bug Fix
- [ ] 📦 Chore (Release)
- [ ] ✅ Test

## Related Tickets & Documents
https://hous-bssb.atlassian.net/browse/BPHH-1682

## Steps to QA
- Difficult to manually test. Posting a video, when a missing pdf was forcefully produced via code

https://github.com/bcgov/HOUS-permit-portal/assets/32022335/89dfba68-a006-46cd-a509-6e60c4f24445

### CODE REVIEW

File: app/frontend/services/api/index.ts [hunk 1]
```diff
@@ -224,10 +224,12 @@ export class Api {
     })
   }
 
+  async generatePermitApplicationMissingPdfs(id: string) {
+    return this.client.post<never>(`/permit_applications/${id}/generate_missing_pdfs`)
+  }
+
   async submitPermitApplication(id, params) {
```
Comment (bkmaibach) [inline app/frontend/services/api/index.ts:231]: remove params if no longer needed?

File: app/policies/permit_application_policy.rb [hunk 2]
```diff
@@ -32,6 +32,10 @@ def submit?
     update?
   end
 
+  def generate_missing_pdfs?
+    true
```
Comment (bkmaibach) [inline app/policies/permit_application_policy.rb:36]: shouldnt there be some check that the user has the right to see this PDF

Comment (taw27) [reply app/policies/permit_application_policy.rb:36]: @bkmaibach you are right. My initial assumption was all authenticated user should be able to generate this permit application once submitted. But there should be a check that submitters own the permit application and reviewers belong to the same jurisdiction. Good catch

File: app/frontend/services/api/index.ts (hunk 1 above)
Comment (taw27) [reply app/frontend/services/api/index.ts:231]: was mistakenly removed from `submitPermitApplication` so putting it back

**Review by bkmaibach — APPROVED** [review]

**Pull request merged**